static/client.js

```diff
@@ -823,34 +829,42 @@ function toggle_panel(x) {
             $('#search_youtube_input').focus();
     }
 }
-//TODO
+
 function search_youtube() {
-    $.ajax({
-        type: "GET",
-        url: "http://gdata.youtube.com/feeds/api/videos",
-        data: {v: 2, "max-results": 20, alt: "jsonc", format: 5, q: $("#search_youtube_input").val()},
-        dataType: "jsonp",
-        success: function (data) {
-            if(data.data != undefined && data.data.items != undefined) {
-                var items = data.data.items;
+	send_event('search_youtube', {
+		query: $("#search_youtube_input").val(), 
+		client: yourSession
+	});
+	show_search_message("Loading...");
+}
+
+function show_search_message(message) {
+                var html = "<span class='notice'>" +
+			message +
+			"</span>";
+                $("#search_youtube_results").html(html);
+}
+
+function show_search_results(data) {
+	if (data == undefined || data.data == undefined || data.data.length < 1) {
+		show_search_results("No results found.");
+	}
+	else {
                 var html = "<table><tr>"; //oh god I am so sorry it came to this
-                for(var i in items) {
-                    var item = items[i];
-                    var title = $('<span>').text(item.title).html(),
-                        thumbnail = item.thumbnail.sqDefault,
-                        id = item.id;
-                    html += "<td><a href=\"#\" onclick=\"set_video('" + id + "');return false\" class=\"pane_item youtube_result\">" +
-                        "<div style=\"text-align:center\">" +
-                        "<img src=\"" + thumbnail + "\">" +
-                        "</div>" +
-                        title + "</a></td>";
+                for(var i in data.data) { 
+                	var title = "<span>" + data.data[i].title + "</span>";
```

Comment:
Do not blindly concatenate HTML with text strings. The original implementation of this escaped HTML by calling jQuery's text(str) then html(). This code is subject to HTML injections.